test_app/tests/authentication/utils/test_claims_reconcile_user.py

```diff
@@ -258,5 +259,10 @@ def test_remove_users_setting_set(org_member_rd, org_admin_rd, member_rd, admin_
     assert user.has_obj_perm(teams[1], 'change')
 
     # Nothing was specified in orgs[2], teams[2]
-    assert not user.role_assignments.filter(object_id=str(orgs[2].id)).exists()
-    assert not user.role_assignments.filter(object_id=str(teams[2].id)).exists()
+    org_content_type = ContentType.objects.get_for_model(Organization)
+    user_role_assignments_for_org_2 = user.role_assignments.filter(content_type=org_content_type, object_id=str(orgs[2].id))
+    assert (
+        not user_role_assignments_for_org_2.exists()
+    ), f"Expected no assignments for org2 but got {', '.join([r.role_definition.name for r in user_role_assignments_for_org_2])}"
+    team_content_type = ContentType.objects.get_for_model(Team)
+    assert not user.role_assignments.filter(content_type=team_content_type, object_id=str(teams[2].id)).exists()
```

Comment:
There was something funky going on with this test after our changes. We were unable to replicate. What we found in our debugging (partially from the change above) was that another Team that the user had permissions to had the same object_id as `orgs[2]` so the assert was constantly failing. In the end, to get the test back to green we added a `content_type` to the filter so that we were only looking for the ID and the specific ContentType along with it.